frame/evm/src/lib.rs

```diff
@@ -677,6 +678,29 @@ pub mod pallet {
 
 	#[pallet::storage]
 	pub type Suicided<T: Config> = StorageMap<_, Blake2_128Concat, H160, (), OptionQuery>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			let pov_before = get_proof_size();
+
+			let zero_account: H160 = H160::zero();
+
+			// just a dummy read to populate the pov with the intermediates nodes
+			let _ = AccountCodesMetadata::<T>::get(zero_account.clone());
+			let (_, min_gas_weight) = T::FeeCalculator::min_gas_price();
+			let (_, account_basic_weight) = Pallet::<T>::account_basic(&zero_account);
+
+			let pov = get_proof_size().unwrap_or_default() - pov_before.unwrap_or_default();
+
+			let total_weight = Weight::from_parts(0, pov)
+				+ T::DbWeight::get().reads(1)
+				+ account_basic_weight
+				+ min_gas_weight;
```

Comment:
Not really important in this case, but we should always use `safe` arithmetic.

```suggestion
			let total_weight = T::DbWeight::get().reads(1)
				.saturating_add(Weight::from_parts(0, pov))
				.saturating_add(account_basic_weight)
				.saturating_add(min_gas_weight);
```